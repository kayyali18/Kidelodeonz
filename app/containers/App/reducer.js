/*
 * AppReducer
 *
 * The reducer takes care of our data. Using actions, we can change our
 * application state.
 * To add a new action, add it to the switch statement in the reducer function
 *
 * Example:
 * case YOUR_ACTION_CONSTANT:
 *   return state.set('yourStateVariable', true);
 */

import { fromJS } from 'immutable';

import { LOAD_API_SUCCESS, LOAD_API, LOAD_API_ERROR } from './constants';

// The initial state of the App
const initialState = fromJS({
  loading: false,
  error: false,
  tasteDive: {
    apiData: false,
  },
  searchQuery: ''
});

function appReducer(state = initialState, action) {
  switch (action.type) {
    case LOAD_API:
      return state
        .set('loading', true)
        .set('error', false)
        .setIn(['tasteDive', 'apiData'], false)
    case LOAD_API_SUCCESS:
      return state
        .setIn(['tasteDive', 'apiData'], action.data)
        .set('loading', false)
    case LOAD_API_ERROR:
      return state.set('error', action.error).set('loading', false)
    default:
      return state
  }
}

export default appReducer
