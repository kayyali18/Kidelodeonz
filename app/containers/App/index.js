/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react'
import { Helmet } from 'react-helmet'
import styled from 'styled-components'
import { Switch, Route } from 'react-router-dom'

import HomePage from 'containers/HomePage/Loadable'
import NotFoundPage from 'containers/NotFoundPage/Loadable'
import Footer from 'components/Footer'
import Header from 'components/Header'
import GlobalStyle from '../../global-styles'

const AppWrapper = styled.div`
  display: grid;
  grid-template-rows: 1fr 8fr 1fr;
  height: 100vh;
  background: linear-gradient(#ff700b, #ff275f);
`

export default function App() {
  return (
    <AppWrapper>
      <Helmet titleTemplate="%s - Kidelodeon" defaultTitle="Kidelodeon">
        <meta
          name="A kid friendly application for kids to explore the web"
          content="Kidelodeon Fumble"
        />
      </Helmet>
      <Header />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/about" component={HomePage} />
        <Route exact path="/faq" component={HomePage} />
        <Route exact path="/contact" component={HomePage} />
        <Route component={NotFoundPage} />
      </Switch>
      <Footer />
      <GlobalStyle />
    </AppWrapper>
  )
}
