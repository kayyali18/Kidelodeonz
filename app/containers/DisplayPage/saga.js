/**
 * Gets the stumble from the tasteDive API
 */

import { put, select, takeLatest } from 'redux-saga/effects'

import { makeSelectData } from 'containers/App/selectors'
import { RUN_STUMBLE_SAGA } from './constants'
import { displayStumble } from './actions'

/**
 * Stumbling Saga --> Tells the story of how we display our info
 */
export function* getStumble() {
  // Select data from store
  let stumble
  const num = Math.floor(Math.random() * 6) + 1
  const stumbles = yield select(makeSelectData())
  if (stumbles) stumble = stumbles[num]
  else stumble = false
  yield put(displayStumble(stumble))
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* stumblingSaga() {
  // Watches for DISPLAY_STUMBLE actions and calls getStumble when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(RUN_STUMBLE_SAGA, getStumble)
}
