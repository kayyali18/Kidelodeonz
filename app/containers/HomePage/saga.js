/**
 * Gets the suggestions from the tasteDive API
 */

import { call, put, select, takeLatest } from 'redux-saga/effects'
import { LOAD_API } from 'containers/App/constants'
import { apiLoaded, apiLoadingError } from 'containers/App/actions'

import request from 'utils/request'
import { makeSelectQuery } from 'containers/HomePage/selectors'
import { tasteDive } from '../../utils/API_keys'

/**
 * Taste Dive api request/response handler
 */
export function* getSuggestions() {
  // Select query from store
  const key = tasteDive
  const query = yield select(makeSelectQuery())
  const requestURL = `https://tastedive.com/api/similar?k=${key}&info=1&limit=100&q=${query}`

  try {
    // Call our request helper (see 'utils/request')
    const results = yield call(request, requestURL)
    yield put(apiLoaded(results))
  } catch (err) {
    yield put(apiLoadingError(err))
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* githubData() {
  // Watches for LOAD_API actions and calls getSuggestions when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(LOAD_API, getSuggestions)
}
