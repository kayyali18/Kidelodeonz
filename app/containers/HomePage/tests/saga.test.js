/**
 * Tests for HomePage sagas
 */

//  import { put, takeLatest } from 'redux-saga/effects'

//  import { LOAD_API } from 'containers/App/constants'
import { apiLoaded, apiLoadingError } from 'containers/App/actions'

//  import tasteDiveData,
import { put } from 'redux-saga/effects'
import { getSuggestions } from '../saga'

const query = 'cars'

/* eslint-disable redux-saga/yield-effects prettier-ignore */
describe('getSuggestions Saga', () => {
  let getSuggestionsGenerator

  // Because we test twice for a succesful and unsuccessful api load
  // we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getSuggestionsGenerator = getSuggestions()

    // describe the selector (makeSelectQuery)
    const selectDescriptor = getSuggestionsGenerator.next().value
    expect(selectDescriptor).toMatchSnapshot()

    const callDescriptor = getSuggestionsGenerator.next(query).value
    expect(callDescriptor).toMatchSnapshot()
  })

  it('should dispatch the apiLoaded action if it requests the data succesfully', () => {
    const response = {
      data: 'this is mockdata',
    }
    const putDescriptor = getSuggestionsGenerator.next(response).value
    expect(putDescriptor).toEqual(put(apiLoaded(response)))
  })
})
