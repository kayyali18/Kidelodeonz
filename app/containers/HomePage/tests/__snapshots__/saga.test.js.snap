// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSuggestions Saga should call the apiLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getSuggestions Saga should call the apiLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?k=322753-kidelode-CRQ0TIZQ&info=1&limit=100&q=cars",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getSuggestions Saga should dispatch the apiLoaded action if it requests the data succesfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getSuggestions Saga should dispatch the apiLoaded action if it requests the data succesfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?k=322753-kidelode-CRQ0TIZQ&info=1&limit=100&q=cars",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
